internal/commands/server/k8s_e2e_test.go

```diff
@@ -934,13 +1011,14 @@ func checkTCPTLSRoute(t *testing.T, port int, expected string, message string) {
 		if err != nil {
 			return false
 		}
-		tlsConn := tls.Client(conn, &tls.Config{
-			InsecureSkipVerify: true,
-		})
+		tlsConn := tls.Client(conn, config)
 		data, err := io.ReadAll(tlsConn)
+
 		if err != nil {
-			return false
+			t.Log(err)
+			return strings.HasPrefix(err.Error(), expected)
 		}
```

Comment:
Should the other `check*` functions be updated similarly to also be able to check for expected error states?

Reply:
Originally the `check*` functions didn't take into account errors, only passing if no error was there, i.e. has `require.NoError` implicit in it. I'd just keep it the way it is for now unless you are actually touching them in the updated tests. We can always do a subsequent refactor down the line.